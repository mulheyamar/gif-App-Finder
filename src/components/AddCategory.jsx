import { useState } from 'react';


export const AddCategory = ({ onNewCategory }) => {
    
    const [ inputValue, setInputValue ] = useState('')
    
    //función para capturar en inputValue las pulsaciones del teclado
    const onInputChange = ({ target }) => {        
        setInputValue( target.value );
    }

    //función que envía lo que escribimos en el formulario y actualiza el listado
    const onSubmit = ( event ) => {
        event.preventDefault();//evita que  la página se recargue y que el formulario no se envíe para poder validar los datos del formulario.
        const newInputValue = inputValue.trim()
        //1ª validación:
        if( newInputValue.length <= 1) return; //sale de la función si el valor sin espacios delante y detrás es igual o menor a uno.

        //actualizamos listado:
        onNewCategory( newInputValue )
        //setCategories( categories => [ inputValue,...categories  ]);

        //limpiamos la caja del input
        setInputValue('');
       
    }

  return (
    <form  onSubmit = { onSubmit }> 
         
        <input
            type = "text"
            placeholder="Buscar gifs"
            value = { inputValue }
            onChange={ onInputChange }
        />

    </form>
   
   // el form hace un refresco de la web al pulsar enter
   
  )
}
