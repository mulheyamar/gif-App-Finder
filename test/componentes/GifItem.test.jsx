import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem/>', () => { 
const title = 'trump foto'
        const url = 'https://media0.giphy.com/media/l2JhIUyUs8KDCCf3W/giphy.gif?cid=41f37cafudkv0fnfq1lg7mi6c6bwic4xnlq8ujpaiqi74cjz&ep=v1_gifs_search&rid=giphy.gif&ct=g'
    test('Debe de coincidir el snapshot', () => { 
        

        const { container } = render( <GifItem title= { title } url={ url }/>);

          expect( container ).toMatchSnapshot();

     });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render( <GifItem title= { title } url={ url }/>);
        //screen.debug();
        // expect( screen.getByRole('img').src).toBe(url);
        // expect( screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });

    test('Debe de mostrar el tÃ­tulo en el componente', () => { 
        render( <GifItem title= { title } url={ url }/>);
        expect( screen.getByText( title )).toBeTruthy();

     })

 });