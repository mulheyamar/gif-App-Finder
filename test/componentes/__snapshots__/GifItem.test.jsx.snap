// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem/> Debe de coincidir el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="trump foto"
      src="https://media0.giphy.com/media/l2JhIUyUs8KDCCf3W/giphy.gif?cid=41f37cafudkv0fnfq1lg7mi6c6bwic4xnlq8ujpaiqi74cjz&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      trump foto
    </p>
  </div>
</div>
`;
